cpp/daal/include/services/internal/buffer.h

```diff
@@ -120,50 +120,93 @@ class Buffer : public Base
      *  \param[out] status  Status of operation
      *  \return host-allocated shared pointer to the data
      */
-    SharedPtr<T> toHost(const data_management::ReadWriteMode & rwFlag, Status * status = NULL) const
+    SharedPtr<T> toHost(const data_management::ReadWriteMode & rwFlag, Status & status) const
     {
         if (!_impl)
         {
-            internal::tryAssignStatusAndThrow(status, ErrorEmptyBuffer);
+            status |= ErrorEmptyBuffer;
             return SharedPtr<T>();
         }
-
         return internal::HostBufferConverter<T>().toHost(*_impl, rwFlag, status);
     }
 
+#ifndef DAAL_NOTHROW_EXCEPTIONS
+    /**
+     *  Converts data inside the buffer to the host side, throws exception if conversion fails
+     *  \param[in]  rwFlag  Access flag to the data
+     *  \return host-allocated shared pointer to the data
+     */
+    SharedPtr<T> toHost(const data_management::ReadWriteMode & rwFlag) const
```

Comment:
What's the reason to create throwable overloading? I suppose we always need to use the version with status parameter on the library side since we do not know will be exceptions enabled on the user side or not.

Reply:
discussed offline - agreed to add throwIfPossible in the "status" version of `toHost()` in next PR